kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -72,55 +72,118 @@ export default function SubscriptionForm() {
             });
     }, [formStep, stripeLoadingPromise]);
 
-    function openStripeModal() {
-        if (!stripeLoadingPromise) {
-            return;
+    const STRIPE_CONTINUE_SESSIONSTORAGE_KEY = 'stripe-form-continue';
+
+    /**
+     * If you arrived on this page, being anonymous, you'd have to first sign in.
+     * Suppose that you do that, we will make sure to send you back to this page
+     * with the sessionStorage key set.
+     * Basically, if you have that sessionStorage key, it will, for you, check the
+     * checkbox and press the "Continue" button.
+     */
+
+    useEffect(() => {
+        if (userData && userData.isAuthenticated) {
+            let autoTriggerStripe = false;
```

Comment:
can we move that sessionStorage value extraction & deletion into a separate function? I think it would aid readability.

Reply:
Could look like this:
```javascript
useEffect(() => {
    if (userData && userData.isAuthenticated && popStripeContinuation()) {
        setPaymentAuthorized(true);
        setOpenStripeModal(true);
    }
}, [userData]);
```
(there's probably a better name somewhere out there)